core/src/main/java/hudson/model/User.java

```diff
@@ -1066,6 +1066,7 @@ private static class Replacer {
         }
 
         private Object readResolve() {
+            // Will generally only work if called after UserIdMapper.init:
```

Comment:
Consistent with the comment in https://github.com/jenkinsci/jenkins/blob/86786b035c226276792068c66d0a8ae7468d33b6/core/src/main/java/hudson/model/AbstractItem.java#L643 (which I did _not_ write).